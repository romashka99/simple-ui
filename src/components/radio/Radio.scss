@import '../../styles/colors.scss';
@import '../../styles/main.scss';
@import '../../styles/style.scss';

.radio {
	display: flex;
	align-items: center;
	flex-wrap: wrap;
	position: relative;
	cursor: pointer;

	& .radio--label {
		padding-left: 8px;

		color: $color-grey;

		font-size: 16px;
		line-height: 24px;
	}

	& .radio--inner {
		-webkit-appearance: none;
		box-sizing: border-box;
		outline: none;
		position: relative;
		margin: 0;

		cursor: pointer;

		width: 20px;
		height: 20px;

		background: $color-white;
		border: $border;
		border-radius: $border-radius-circle;

		&:checked:after {
			content: '';
			position: absolute;
			width: 12px;
			height: 12px;
			top: 3px;
			left: 3px;
			background: $color-grey;
			border-radius: $border-radius-circle;
		}

		&:disabled {
			cursor: not-allowed;
			background: $color-light-grey;
			border-color: $color-mid-light-grey;

			&:checked:after {
				background: $color-mid-light-grey;
			}

			& + .radio--label {
				cursor: not-allowed;
				color: $color-mid-light-grey;
			}
		}

		&:hover {
			&:not(:checked) {
				&:not(:disabled) {
				}
			}
		}
	}
}
