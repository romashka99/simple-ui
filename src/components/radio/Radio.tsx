import React, { DetailedHTMLProps, InputHTMLAttributes } from 'react';
import { Color, Size } from '../../types/types';
import './Radio.scss';

interface IRadioProps
	extends DetailedHTMLProps<
			InputHTMLAttributes<HTMLInputElement>,
			HTMLInputElement
		>,
		React.AriaAttributes {
	color?: Color;
	scale?: Size;
	label?: string;
}

export default function Radio({
	color = 'default',
	scale = 'medium',
	label,
	className,
	...props
}: IRadioProps) {
	const classes = `radio radio--${color} radio--${scale}${
		className ? ' ' + className : ''
	}`;

	return (
		<>
			<input data-testid='radio' type='radio' className={classes} {...props} />
			{label && <label>{label}</label>}
		</>
	);
}
