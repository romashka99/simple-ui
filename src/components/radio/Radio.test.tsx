import { render, screen } from '@testing-library/react';
import Radio from '.';

describe('Test Radio', () => {
	test('without attr', () => {
		render(<Radio />);
		const input = screen.getByTestId('radio');
		expect(input.classList.length).toBe(3);
		expect(input.classList.contains('radio')).toBe(true);
		expect(input.classList.contains('radio--medium')).toBe(true);
		expect(input.getAttribute('disabled')).toBeNull();
		expect(input).toMatchSnapshot();
	});
});
