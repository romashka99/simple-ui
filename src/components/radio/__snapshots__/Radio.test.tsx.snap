// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Radio without attr 1`] = `
<input
  class="radio radio--default radio--medium"
  data-testid="radio"
  type="radio"
/>
`;
